### PR TITLE
Shared-library-implementation

### PR DESCRIPTION
**Pull request by ivanslabbert**

### CODE REVIEW

File: app/src/components/content/VideoPlayer.spec.ts [hunk 1]
```diff
@@ -30,12 +31,14 @@ vi.mock("video.js", () => {
 describe("VideoPlayer", () => {
     it("renders the poster image", async () => {
         mount(VideoPlayer, {
-            props: { contentParent: mockPost },
+            props: {
+                content: mockEnglishContentDto as ContentDto,
```
Comment (ivanslabbert) [inline app/src/components/content/VideoPlayer.spec.ts:35]: Casting should not be necessary here?

File: app/src/pages/SingleContent.spec.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        ] as BaseDocumentDto[]);
+
+        setActivePinia(createPinia());
+    });
+
+    afterEach(() => {
+        db.docs.clear();
+    });
+
+    it("displays the loading when no content", async () => {
```
Comment (ivanslabbert) [inline app/src/pages/SingleContent.spec.ts:25]: Test description should read "displays the loading spinner when no content is available"

File: app/src/pages/SingleContent.spec.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        const wrapper = mount(SingleContent, {
+            props: {
+                slug: mockEnglishContentDto.slug,
+            },
+        });
+
+        expect(wrapper.findComponent({ name: "LoadingSpinner" }).exists()).toBe(true);
+    });
+
+    it("displays the image content", async () => {
```
Comment (ivanslabbert) [inline app/src/pages/SingleContent.spec.ts:35]: Description should be "displays the content image"

File: app/src/pages/SingleContent.spec.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                slug: mockEnglishContentDto.slug,
+            },
+        });
+
+        await waitForExpect(() => {
+            expect(wrapper.html()).toContain("test-image.jpg");
+        });
+    });
+
+    it("displays the video content when defined", async () => {
```
Comment (ivanslabbert) [inline app/src/pages/SingleContent.spec.ts:49]: Description could change (for the sake of consistency) to "displays the content video when defined"

File: app/src/pages/SingleContent.spec.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                slug: mockEnglishContentDto.slug,
+            },
+        });
+
+        await waitForExpect(() => {
+            expect(wrapper.html()).toContain("video-player-stub");
+        });
+    });
+
+    it("displays the pusblish date content", async () => {
```
Comment (ivanslabbert) [inline app/src/pages/SingleContent.spec.ts:66]: Typo: "pusblish" should be "publish"

File: app/src/pages/SingleContent.spec.ts [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+                slug: mockEnglishContentDto.slug,
+            },
+        });
+
+        await waitForExpect(() => {
+            expect(wrapper.html()).toContain("Jan 1, 2024");
+        });
+    });
+
+    it("displays the summary content", async () => {
```
Comment (ivanslabbert) [inline app/src/pages/SingleContent.spec.ts:78]: "displays the content summary"

File: app/src/pages/SingleContent.spec.ts [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+                slug: mockEnglishContentDto.slug,
+            },
+        });
+
+        await waitForExpect(() => {
+            expect(wrapper.html()).toContain("This is an example post");
+        });
+    });
+
+    it("displays the text content", async () => {
```
Comment (ivanslabbert) [inline app/src/pages/SingleContent.spec.ts:90]: "displays the content text"

File: app/src/pages/SingleContent.spec.ts [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+                slug: mockEnglishContentDto.slug,
+            },
+        });
+
+        await waitForExpect(() => {
+            expect(wrapper.find("p").exists()).toBe(true);
+        });
+    });
+
+    it("display the categories", async () => {
```
Comment (ivanslabbert) [inline app/src/pages/SingleContent.spec.ts:102]: This test is not only for categories, but for any tags (even though the test is only using a category tag). Think the description should rather be "displays tags"

File: app/src/stores/socketConnection.ts [hunk 9]
```diff
@@ -58,30 +58,30 @@ export const useSocketConnectionStore = defineStore("socketConnection", () => {
             const baseRepository = new BaseRepository();
```
Comment (ivanslabbert) [inline app/src/stores/socketConnection.ts:58]: Lint error

File: app/src/stores/socketConnection.ts [hunk 10]
```diff
@@ -58,30 +58,30 @@ export const useSocketConnectionStore = defineStore("socketConnection", () => {
...
             const groupsPerDocType = accessMapToGroups(accessMap, AclPermission.View);
```
Comment (ivanslabbert) [inline app/src/stores/socketConnection.ts:59]: lint error

File: app/src/stores/socketConnection.ts [hunk 11]
```diff
@@ -58,30 +58,30 @@ export const useSocketConnectionStore = defineStore("socketConnection", () => {
...
+
+            //         // Delete associated Language content documents
+            //         if (docType === DocType.Language) {
+            //             const revokedLanguages = await revokedDocs.toArray();
+            //             const revokedlanguageIds = revokedLanguages.map((l) => l._id);
+            //             await baseRepository.whereLanguageIds(revokedlanguageIds).delete();
+            //         }
+
+            //         await revokedDocs.delete();
+            //     });
```
Comment (ivanslabbert) [inline app/src/stores/socketConnection.ts:84]: Can delete this. Will be replaced with sync code from shared library

**Review by belljohan3 — APPROVED** [review]
Great Job @ivanslabbert

**Pull request merged**